### DISCUSSION TITLE
Issues getting AU lab to show up in microphone section in security & privacy pannel

### DISCUSSION THREAD

**Discussion by AxelAxelAlex** [ExistentialAudio/BlackHole#520] [Q&A]
I followed the guide completely except for the microphone portion since it doesn't work. Naturally, I'm not getting any audio output through blackhole. I'm using an m1 mac mini (early 2020). I've tried connecting my mic & headphones directly through the internal dac as well as an external one. Does someone know what's up?

**Comment by devinroth** [comment]
There’s been a thread on the topic already if you want to search it out.

I think the solution was put the app in the downloads folder then open it.

Devin Roth

**Comment by AxelAxelAlex** [comment]
Yeah I did that. But it still didn't prompt the request for mic access. Are there any other known solutions?

**Comment by devinroth** [comment]
No idea. Maybe there’s a way to manually add it. AULab is old and no longer supported by Apple.

If you figure it out post the answer here as it might help others.

Devin Roth

**Reply by AxelAxelAlex** [reply]
I will once I find a solution. I also didn't get an access prompt from blackhole itself, but I take it this is not the issue?

**Comment by devinroth** [comment]
BlackHole doesn’t require microphone access.

Devin Roth

**Reply by AxelAxelAlex** [reply]
https://superuser.com/questions/1441270/apps-dont-show-up-in-camera-and-microphone-privacy-settings-in-macbook

This is a useful link for people with the same issue, it didn't work for me though. I'm going to try manually giving it access with the command 
tccctl --grant -b Aulab bundle id -s kTCCServiceMicrophone 
but I don't know how to find a bundle ID of an app I've been searching for ages now.

**Reply by AxelAxelAlex** [reply]
I might as well just reset my mac, but do you think that'll solve the issue?

**Reply by AxelAxelAlex** [reply]
Damn I reset my entire mac just to have the same problem... I'm at a loss

**Comment by mgeorgetw** [comment]
You are close trying to grant access of microphone from command line. [This reddit thread](https://www.reddit.com/r/oratory1990/comments/qi8s6h/comment/j7un2hw/?utm_source=share&utm_medium=web2x&context=3) solved it for me. Here's what I did:

1. Go to System Settings > Privacy & Security > Grant Full Disk Access to Terminal

2. Go to Terminal app and open Apple's TCC (Transparency, Consent, and Control) database with SQLite3 using this command (Replace <Username> with your user name): 
`sqlite3 /Users/<Username>/Library/Application\ Support/com.apple.TCC/TCC.db`

3. Insert a row that grant microphone access to AU Lab with this command:
`INSERT INTO access VALUES('kTCCServiceMicrophone', 'com.apple.audio.aulab', 0, 2, 2, 1, X'fade0c000000003400000001000000060000000200000015636f6d2e6170706c652e617564696f2e61756c616200000000000003', NULL, NULL, 'UNUSED', NULL, 0, CAST(strftime('%s', 'now') AS INTEGER), NULL, NULL, 'UNUSED', 0);` 

4. Exit SQLite3 with `CTRL + d`
5. Go to System Settings > Privacy & Security > Microphone. See whether AU Lab is in the list with permission granted.
6. Go to System Settings > Privacy & Security > Full Disk Access. Uncheck Terminal.

Works for me with macOS Sonoma.

**Reply by Seifi-Hifi** [reply]
you are a real magician

**Comment by ghost** [comment]
Hello,

I posted a solution to this in a different thread. I might as well post it here in case it helps:


https://youtu.be/2RigWfQ4lGU?si=vzzt7r2aOVRQ-0j8


Cheers!

**Comment by Bushwickz** [comment]
I know this is old, but I want to clarify @mgeorgetw’s comment for those dealing with the standard 13-column SQLite table on Ventura. After completing steps 1 and 2, in step 3, insert this exact command, then save and exit with .exit. Then continue with steps 5 and 6. 

On macOS Ventura, you might not see the Microphone permission update immediately in Privacy & Security, but the change takes effect as soon as you exit SQLite using .exit.

On that note, Backing up the TCC.db file is crucial because it stores macOS application permissions for sensitive resources like the microphone, camera, and full disk access. If you modify these permissions—especially manually—you risk corrupting the database. The system-wide TCC.db is protected by System Integrity Protection (SIP), making it difficult to modify. Having a backup allows you to revert changes without needing to disable SIP or perform complex recovery steps.



```
INSERT INTO access VALUES(
  'kTCCServiceMicrophone',
  'com.apple.audio.aulab',
  0,
  2,
  2,
  1,
  X'fade0c000000003400000001000000060000000200000015636f6d2e6170706c652e617564696f2e61756c616200000000000003',
  NULL,
  NULL,
  'UNUSED',
  NULL,
  0,
  CAST(strftime('%s', 'now') AS INTEGER)
);
```

**Comment by Atalantia** [comment]
@Bushwickz
This is the way to go. I think you have to deactivate SIP on Monterey otherwise you can't modifier the TCC.db file
